### ISSUE TITLE
Can't detect Text-Lines, LinkRefiner from CRAFT might solve the issue

### ISSUE THREAD

**Issue by deepseek** [MhLiao/DB#82]
.

**Comment by MhLiao** [comment]
@deepseek Are your labels in the text-line level?
It seems that there are the same shifts for all the instances. I suggest you check the labels. 
Moreover, if you did not use DCN, you can try other tricks to enlarge the receptive field, such as PPM.

**Comment by deepseek** [comment]
@MhLiao 
- Yes, my labels are in the text-line level, they are synthetically generated.
- At training, I used `totaltext_resnet18_deform_thre.yaml`, isn't DCN used by default?
- How can I use PPM `pyramid pooling module`?

**Comment by MhLiao** [comment]
@deepseek It is trained with DCN. But I am curious about why your results are shifted.

**Comment by deepseek** [comment]
@MhLiao Sorry for being late,
I re-trained the model again using more pages from the dataset, currently I have reached:
`step: 112000, epoch: 2352` and still DB can't detect the text-lines even at `--box_thresh 0.0`
[The used text-line dataset, along with the latest model.
](https://drive.google.com/drive/folders/1dD6Hewe6wUpgjH0NlbW2a_KHAQoKUQl2?usp=sharing)

**Comment by deepseek** [comment]
Still training, I have reached `step:142000, epoch: 2784`, and DB still can't detect lines
![out](https://user-images.githubusercontent.com/49877135/71354554-5d0d7180-258d-11ea-94f6-780f35dfb1da.jpg)

**Comment by MhLiao** [comment]
@deepseek I think there are something wrong. Have you checked the ground-truth map in the training period? To debug it, you can train the model with only a single image to see whether the model can overfit it or not.

**Comment by deepseek** [comment]
- You are smart, [the first image that I posted](https://github.com/MhLiao/DB/issues/82#issue-539886344) was me training on a single image to test and see if `DB` can overfit on text-lines, `DB` failed to overfit.
- Later-on, [after your suggestion](https://github.com/MhLiao/DB/issues/82#issuecomment-567208289), I trained using the entire dataset to test if there is an issue with the labels, but still `DB` can't detect text-lines.
[The dataset, model and logs are uploaded here.
](https://drive.google.com/drive/folders/1dD6Hewe6wUpgjH0NlbW2a_KHAQoKUQl2?usp=sharing)
Conclusion:
- `DB` seems to fail on detecting text-lines.

**Comment by MhLiao** [comment]
@deepseek 
If the model can not overfit a single image, there must be something wrong with the optimization (either the target labels or the loss function). 
I strongly suggest you disable the data augmentation to double-check it by training with a single image. 
Another potential issue may be the interpolation types of up-sample operators, which may result in a shift in the feature map. Maybe you can change them to bi-linear.

**Comment by deepseek** [comment]
@MhLiao 
What exact code should be changed to disable data augmentation and enable bilinear interpolation.
Can you test the dataset that I uploaded.

**Comment by deepseek** [comment]
@MhLiao

**Comment by deepseek** [comment]
@MhLiao remember :sunglasses:

**Comment by titikid** [comment]
> @MhLiao
> DB detects words well, but not text-lines, thats why using a technique like the one created by CRAFT, which is called LinkRefiner, would combine the detected words into lines.

Which configuration did you use for word-based detection? i checked pretrained models and found that ic15 model is good for word-based and td500 model is better for line-based.
@MhLiao : what is final loss and how many epoch when you training with ic15?

**Comment by deepseek** [comment]
@titikid Have you tried training text-lines yourself?
is there much difference between td500 and ic15 config?

**Comment by titikid** [comment]
> @titikid Have you tried training text-lines yourself?
> is there much difference between td500 and ic15 config?

I tried word-based training using ic15 config and it works well.
i haven't try text-lines training, but pretrained model by td500 was quite good

**Comment by eyebies** [comment]
Did it work on the 
![vXEYa](https://user-images.githubusercontent.com/3131247/75080142-cd942880-5513-11ea-9d31-27ad0716a230.jpg)

**Comment by deepseek** [comment]
Then try training text-lines and report back

**Comment by eyebies** [comment]
I was able to fine-tune my dataset (with long text) from the td500_resnet50 weights (not resnet18). I did not change a lot of parameters in the yml

**Comment by deepseek** [comment]
@eyebies @titikid 
When I tested the pretrained model `td500_resnet50` the results on the td500 images were bad. look at the image below.
So I tried training my own model using the `td500_resnet50_deform_thre.yaml` as you suggested, but after hours and hours of training, the loss is still at 4.7-5 and it's very bad.
I am using the default yaml configuration but changes the batches number.
![IMG_0504](https://user-images.githubusercontent.com/49877135/75118547-1a9b0a80-568c-11ea-96aa-b352fef6cb39.jpg)
